third_party/kiva/cmake/compiler-flags.cmake

```diff
@@ -4,6 +4,18 @@ add_library(kiva_common_interface INTERFACE)
   # Compiler flags #
   #================#
 
+
+cmake_policy(PUSH)
+# This is needed to have <CXX_COMPILER_ID> differentiate between Clang and AppleClang
+cmake_policy(SET CMP0025 NEW)
+
+# AppleClang version numbering does not follow the upstream LLVM clang version.
+# In order to properly silence the -Wenum-constexpr-conversion warning without adding another
+# warning for "unknown warning option", we need to add this only for versions that support this flag.
+set(clang_16_or_greater "$<AND:$<CXX_COMPILER_ID:Clang>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,16>>")
+set(apple_clang_15_or_greater "$<AND:$<CXX_COMPILER_ID:AppleClang>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,15>>")
+set(has_enum_constexpr_option "$<OR:${clang_16_or_greater},${apple_clang_15_or_greater}>")
```

Comment:
Ok, this is nicely deconstructed and clear. Thanks!

----

As a side note: I wonder at which point `CheckCXXCompilerFlag` makes more sense, but whatever, this is fine! 

https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html

Also, CMake should ideally have some kind of genex for this. `$<ADD_FLAG_IF_SUPPORTED:-Wno-enum-constexpr-conversion>`. I added a feature request for this on https://gitlab.kitware.com/cmake/cmake/-/issues/25198